fix(platform-server): remove dependency on `@angular/platform-browser-dynamic`

alan-agius4 opened the pull request:
Using `@angular/platform-browser-dynamic` is no longer required for JIT scenarios with Ivy. Instead `@angular/compiler` should be imported instead.

This change is part of the effort to reduce the server bundles sizes, which is needed to support cloud workers.

BREAKING CHANGE:
Users that are using SSR with JIT mode will now need to add  `import to @angular/compiler` before bootstrapping the application.

**NOTE:** this does not effect users using the Angular CLI.

AndrewKushnir reviewed (approved):
Reviewed-for: public-api

@alxhub could you please also take a look at the change before the merge?

alan-agius4 committed:
fix(platform-server): remove dependency on `@angular/platform-browser-dynamic`

Using `@angular/platform-browser-dynamic` is no longer required for JIT scenarios with Ivy. Instead `@angular/compiler` should be imported instead.

This change is part of the effort to reduce the server bundles sizes, which is needed to support cloud workers.

BREAKING CHANGE:
Users that are using SSR with JIT mode will now need to add  `import to @angular/compiler` before bootstrapping the application.

**NOTE:** this does not effect users using the Angular CLI.

jessicajaniuk reviewed (approved):
reviewed-for: public-api

alan-agius4 commented:
**Caretaker note:** [Presubmit](http://fusion2/presubmit/527855641/OCL:527855641:BASE:527855664:1682684468158:1e939bc3/targets) was green prior to the commit message change.

alxhub commented:
This PR was merged into the repository by commit 06452af31fb741c3d2ba8e653e1ca830f27960a8.
